analyses/independent-samples/00-repeated-samples.Rmd

```diff
@@ -0,0 +1,221 @@
+
+## Unlabeled composition samples
+
+9 samples are listed as "Not Reported" for `composition`. 
+What are these, and do we have correponding samples for which composition is known?
+
+```{r}
+notreported_participants <- seq_samples %>%
+  filter(composition == "Not Reported") %>%
+  pull(Kids_First_Participant_ID) %>%
+  unique() 
+
+seq_samples %>% 
+  filter(Kids_First_Participant_ID %in% notreported_participants)
+```
+
+In one case (`PT_N8W26H19`), we do have Solid Tissue samples, but not for the others.
+Should these samples be included in further analysis?
+
+## WGS vs WXS
+```{r}
+WXS_participants <- seq_samples %>%
+  filter(experimental_strategy == "WXS") %>%
+  pull(Kids_First_Participant_ID) %>%
+  unique() 
+
+WGS_participants <- seq_samples %>%
+  filter(experimental_strategy == "WGS") %>%
+  pull(Kids_First_Participant_ID) %>%
+  unique() 
+
+No_WGS <- WXS_participants[!(WXS_participants %in% WGS_participants)]
+
+length(No_WGS)
+```
+
+So there are currently 16 samples with missing WGS data, but that should change in a future release.
+
+Just to check, is this still true when we restrict to early samples?
+```{r}
+WXS_participants <- seq_samples %>%
+  filter(experimental_strategy == "WXS", 
+         Kids_First_Biospecimen_ID %in% early_samples) %>%
+  pull(Kids_First_Participant_ID) %>%
+  unique() 
+
+WGS_participants <- seq_samples %>%
+  filter(experimental_strategy == "WGS", 
+         Kids_First_Biospecimen_ID %in% early_samples) %>%
+  pull(Kids_First_Participant_ID) %>%
+  unique() 
+
+No_WGS <- WXS_participants[!(WXS_participants %in% WGS_participants)]
+
+length(No_WGS)
+```
+
+No change. For now, we will look only at WGS samples, but this could be adjusted in future analysis.
+
+
+
+## Summarize repeated samples
+
+```{r}
+repeat_samples <- seq_samples %>% 
+  dplyr::filter(Kids_First_Biospecimen_ID %in% early_samples, 
+                experimental_strategy == "WGS") %>%
+  dplyr::filter(composition != "Derived Cell Line") %>%
+  dplyr::filter(Kids_First_Participant_ID %in% 
+                  Kids_First_Participant_ID[duplicated(Kids_First_Participant_ID)])
+
+repeat_samples %>% 
+  group_by(Kids_First_Participant_ID) %>%
+  summarise(samples = paste(Kids_First_Biospecimen_ID, collapse = ", "), 
+         descriptors = paste(sort(unique(tumor_descriptor)),  collapse = ", "),
+         age_diagnosis = paste(sort(unique(age_at_diagnosis_days)), collapse = ", "),
+         experimental_stragegy = paste(sort(unique(experimental_strategy)), collapse = ", ")
+  )
+```
+
+
+## Select independent samples
+
+Using the information described above, I created a function `independent_samples()`
+to select the earliest biospecimen from each participant, or in the case of more 
+than one with the same date, randomly select among the available samples.
+
+An example of this function is below, applied to the WGS Solid Tissue tumor samples.
+Note that in this case, only tumors designated as primary will be included, 
```

Comment:
Can you note what primary means here at the moment (e.g., in v5) in terms of what fields and values you use?